midterm-2023-11-06/code/5094-viktor-lozanovski/script.ts

```diff
@@ -0,0 +1,103 @@
+interface Album {
+  name: string;
+  year: number;
+}
+
+interface Band {
+  id: number;
+  name: string;
+  genre: string;
+  formed: number;
+  location: string;
+  members: string[];
+  albums: Album[];
+}
+
+const bandContainer = document.getElementById('band-container')!;
+
+function renderBands(bands: Band[]) {
+  bandContainer.innerHTML = '';
+
+  bands.forEach((band) => {
+    const bandTable = document.createElement('div');
+    bandTable.classList.add('band-table');
+
+    const bandDataId = document.createElement('div');
+    bandDataId.classList.add('band-data');
+    bandDataId.textContent = band.id.toString();
+    bandTable.appendChild(bandDataId);
+
+    const bandDataName = document.createElement('div');
+    bandDataName.classList.add('band-data');
+    bandDataName.textContent = band.name;
+    bandTable.appendChild(bandDataName);
+
+    const bandDataFormed = document.createElement('div');
+    bandDataFormed.classList.add('band-data');
+    bandDataFormed.textContent = band.formed.toString();
+    bandTable.appendChild(bandDataFormed);
+
+    const bandDataLocation = document.createElement('div');
+    bandDataLocation.classList.add('band-data');
+    bandDataLocation.textContent = band.location;
+    bandTable.appendChild(bandDataLocation);
+
+    const bandDataCountry = document.createElement('div');
+    bandDataCountry.classList.add('band-data');
+    bandDataCountry.textContent = getCountryFromLocation(band.location);
+    bandTable.appendChild(bandDataCountry);
+
+    const bandDataMembers = document.createElement('div');
+    bandDataMembers.classList.add('band-data');
+    bandDataMembers.textContent = getFormattedMembers(band.members);
+    bandTable.appendChild(bandDataMembers);
+
+    const bandDataAlbums = document.createElement('div');
+    bandDataAlbums.classList.add('band-data');
+    bandDataAlbums.textContent = band.albums.length.toString();
+    bandTable.appendChild(bandDataAlbums);
+
+    bandContainer?.appendChild(bandTable);
+  });
+}
+
+fetch('https://raw.githubusercontent.com/sweko/uacs-internet-programming-exams/main/midterm-2023-11-06/data/bands.json')
```

Comment:
Loading the data is done in a nice way, albeit without the async/await pattern. However, it will actually execute twice, since we have the code two times. +5